pkg/odo/util/pushtarget/pushtarget.go

```diff
@@ -0,0 +1,45 @@
+package pushtarget
+
+import (
+	"os"
+
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/preference"
+)
+
+// env variables
+const (
+	// Setting this env to `docker` will enable pushing to docker containers
+	// and will override the setting in the preferences file.
+
+	OdoPushTarget = "ODO_PUSHTARGET"
+)
+
+// IsPushTargetDocker checks if the push target preference has been set to docker
+func IsPushTargetDocker() bool {
```

Comment:
Maybe pass along a filesystem. That way we can use the filesystem in our tests and speed it up.

Reply:
@mik-dass How would the filesystem be passed in? `preference.New()` (which `IsPushTargetDocker` calls) doesn't take in any args, so the preferences package would need to be refactored quite a bit to support `filesystem.Filesystem`

Reply:
Yeah you're right. Sorry about this, I didn't realize we don't have support of fake filesystems for preferences.